src/enclave/enclave_oe.c

```diff
@@ -73,19 +82,69 @@ void sgxlkl_ethread_init(void)
     return;
 }
 
+static void _copy_shared_memory(const sgxlkl_config_t* config_on_host)
+{
+    sgxlkl_enclave_config_shared_memory_t* shm =
+        &sgxlkl_enclave_state.enclave_config->shared_memory;
+    shm->num_virtio_blk_dev = config_on_host->num_disks;
+    shm->virtio_blk_dev_mem =
+        oe_calloc(config_on_host->num_disks, sizeof(void*));
+    for (size_t i = 0; i < config_on_host->num_disks; i++)
+        shm->virtio_blk_dev_mem[i] =
+            config_on_host->disks[i].virtio_blk_dev_mem;
+
+    memcpy(
+        &sgxlkl_enclave_state.enclave_config->shared_memory,
+        &config_on_host->shared_memory,
+        sizeof(sgxlkl_shared_memory_t)); // CHECK: wrong type
+}
+
+static int _read_eeid_config(const sgxlkl_config_t* config_on_host)
+{
+    const oe_eeid_t* eeid = (oe_eeid_t*)__oe_get_eeid();
+    const char* app_config_json = (const char*)eeid->data;
+
+    if (sgxlkl_read_config_json(
+            app_config_json,
+            &sgxlkl_enclave_state.enclave_config,
+            &sgxlkl_enclave_state.app_config))
+        return 1;
+
+    _copy_shared_memory(config_on_host);
+
+    // This will be removed once shared memory and config have been
+    // separated fully.
+    sgxlkl_enclave = sgxlkl_enclave_state.enclave_config;
+
+    return 0;
+}
+
 int sgxlkl_enclave_init(const sgxlkl_config_t* config_on_host)
 {
     SGXLKL_ASSERT(config_on_host);
 
+    sgxlkl_enclave_state.disk_state = NULL;
+
+    sgxlkl_verbose = 0;
+
+#ifndef OE_WITH_EXPERIMENTAL_EEID
+    if (!config_on_host->app_config_str)
```

Comment:
@prp Should I keep the non-EEID path for enclave configuration for debugging purposes, or can I kick this and make it json-enclave-config-only (and add a flag to skip attestation in debug mode)?

Reply:
@wintersteiger I think that the old way of handling configuration inside the enclave can go.